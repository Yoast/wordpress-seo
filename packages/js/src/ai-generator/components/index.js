export { App } from "./app";
export { GoogleContent } from "./google-content";
export { GooglePreview } from "./google-preview";
export { GooglePreviewSkeleton } from "./google-preview-skeleton";
export { Introduction } from "./introduction";
export { LengthProgressBar } from "./length-progress-bar";
export { ModalContent } from "./modal-content";
export { PreviewModePicker } from "./preview-mode-picker";
export { SocialContent } from "./social-content";
export { SocialPreview } from "./social-preview";
export { SocialPreviewSkeleton } from "./social-preview-skeleton";
export { SuggestionError } from "./suggestion-error";
export { SuggestionsList } from "./suggestions-list";
export { SuggestionsListSkeleton } from "./suggestions-list-skeleton";
export { TipNotification } from "./tip-notification";
export { TwitterContent } from "./twitter-content";
export { TwitterPreview } from "./twitter-preview";
export { TwitterPreviewSkeleton } from "./twitter-preview-skeleton";
export { TypeProvider } from "./type-provider";
export { SparksLimitNotification } from "./sparks-limit-notification";
