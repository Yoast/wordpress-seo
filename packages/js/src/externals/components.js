import CollapsibleCornerstone from "../components/CollapsibleCornerstone";
import KeywordInput from "../components/contentAnalysis/KeywordInput";
import ReadabilityAnalysis from "../components/contentAnalysis/ReadabilityAnalysis";
import SeoAnalysis from "../components/contentAnalysis/SeoAnalysis";

window.yoast = window.yoast || {};
window.yoast.externals = window.yoast.externals || {};
window.yoast.externals.components = {
	CollapsibleCornerstone,
	KeywordInput,
	ReadabilityAnalysis,
	SeoAnalysis,
};
