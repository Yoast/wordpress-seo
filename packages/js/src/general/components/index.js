export { AlertsList } from "./alerts-list";
export { AlertsTitle } from "./alerts-title";
export { Collapsible } from "./collapsible";
export { Notice } from "./notice";
export { Notifications } from "./notifications";
export { Problems } from "./problems";
export { RouteErrorFallback } from "./route-error-fallback";
export { RouteLayout } from "./route-layout";
