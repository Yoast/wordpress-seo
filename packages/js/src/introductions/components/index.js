export { Content } from "./content";
export { Introduction } from "./introduction";
export { Modal } from "./modal";
export { IntroductionProvider, useIntroductionsContext } from "./provider";
