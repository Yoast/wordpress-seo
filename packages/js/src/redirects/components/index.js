export { RouteErrorFallback } from "./route-error-fallback";
export { RouteLayout } from "./route-layout";
export { Menu } from "./menu";
export { SidebarLayout } from "./sidebar-layout";
export { Notifications } from "./notifications";
