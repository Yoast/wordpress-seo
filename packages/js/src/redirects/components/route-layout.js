import { useSelect } from "@wordpress/data";
import { __, sprintf } from "@wordpress/i18n";
import { Button, Title } from "@yoast/ui-library";
import PropTypes from "prop-types";
import { LiveAnnouncer, LiveMessage } from "react-aria-live";
import { Helmet } from "react-helmet";
import { useSelectRedirects } from "../hooks";
import { LockClosedIcon } from "@heroicons/react/outline";
import { STORE_NAME } from "../constants";

/**
 * Route layout wrapper with accessible title and description.
 * @param {Object} props The properties.
 * @param {JSX.node} children The children.
 * @param {string} title The title.
 * @param {JSX.node} [description] The description.
 * @returns {JSX.Element} The route layout component.
 */
export const RouteLayout = ( {
	children,
	title,
	description,
} ) => {
	const documentTitle = useSelect( select => select( STORE_NAME ).selectDocumentFullTitle( { prefix: title } ), [] );
	const ariaLiveTitle = sprintf(
		"%1$s - %2$s",
		title,
		"Yoast SEO"
	);

	const upsellLink = useSelectRedirects( "selectLink", [], "https://yoa.st/redirect-manager-upsell" );

	return (
		<LiveAnnouncer>
			<LiveMessage message={ ariaLiveTitle } aria-live="polite" />
			<Helmet>
				<title>{ documentTitle }</title>
			</Helmet>
			<header className="yst-p-8 yst-border-b yst-border-slate-200">
				<div className="yst-max-w-screen-sm">
					<Title>{ title }</Title>
					{ description && <p className="yst-text-tiny yst-mt-3">{ description }</p> }
					<Button
						as="a"
						href={ upsellLink }
						target="_blank"
						variant="upsell"
						size="large"
						className="yst-flex yst-gap-1.5 yst-mt-6 yst-w-fit"
					>
						<LockClosedIcon className="yst-w-4 yst-h-4" />
						<span>{ __( "Unlock with Premium", "wordpress-seo" ) }</span>
					</Button>
				</div>
			</header>
			{ children }
		</LiveAnnouncer>
	);
};

RouteLayout.propTypes = {
	children: PropTypes.node.isRequired,
	title: PropTypes.string.isRequired,
	description: PropTypes.node,
};
