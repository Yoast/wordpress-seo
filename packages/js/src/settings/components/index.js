export { default as FieldsetLayout } from "./fieldset-layout";
export { default as FormLayout } from "./form-layout";
export { default as FormikFlippedToggleField } from "./formik-flipped-toggle-field";
export { default as FormikMediaSelectField } from "./formik-media-select-field";
export { default as FormikReplacementVariableEditorField } from "./formik-replacement-variable-editor-field";
export { default as FormikValueChangeField } from "./formik-value-change-field";
export { default as FormikWithErrorField } from "./formik-with-error-field";
export { default as Notifications } from "./notifications";
export { default as OpenGraphDisabledAlert } from "./open-graph-disabled-alert";
export { default as SidebarNavigation } from "./sidebar-navigation";
export { ReactComponent as YoastLogo } from "./yoast-logo.svg";
