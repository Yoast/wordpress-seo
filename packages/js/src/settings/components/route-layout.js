import { __, sprintf } from "@wordpress/i18n";
import { Title } from "@yoast/ui-library";
import PropTypes from "prop-types";
import { Helmet } from "react-helmet";
import { LiveAnnouncer, LiveMessage } from "react-aria-live";
import { useDocumentTitle } from "../hooks";

/**
 * @param {Object} props The properties.
 * @param {JSX.node} children The children.
 * @param {string} title The title.
 * @param {JSX.node} [description] The description.
 * @returns {JSX.Element} The route layout component.
 */
const RouteLayout = ( {
	children,
	title,
	description,
} ) => {
	const documentTitle = useDocumentTitle( { prefix: `${ title } â€¹ ` } );
	const ariaLiveTitle = sprintf(
		/* translators: 1: Settings' section title, 2: Yoast SEO */
		__( "%1$s Settings - %2$s", "wordpress-seo" ),
		title,
		"Yoast SEO"
	);
	return (
		<LiveAnnouncer>
			<LiveMessage message={ ariaLiveTitle } aria-live="polite" />
			<Helmet>
				<title>{ documentTitle }</title>
			</Helmet>
			<header className="yst-p-8 yst-border-b yst-border-slate-200">
				<div className="yst-max-w-screen-sm">
					<Title>{ title }</Title>
					{ description && <p className="yst-text-tiny yst-mt-3">{ description }</p> }
				</div>
			</header>
			{ children }
		</LiveAnnouncer>
	);
};

RouteLayout.propTypes = {
	children: PropTypes.node.isRequired,
	title: PropTypes.string.isRequired,
	description: PropTypes.node,
};

export default RouteLayout;
