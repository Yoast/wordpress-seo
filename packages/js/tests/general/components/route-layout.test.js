import { render } from "../../test-utils";
import { RouteLayout } from "../../../src/general/components/route-layout";

describe( "RouteLayout", () => {
	const props = {
		children: <>empty component </>,
		title: "Title",
		description: "Description",
	};

	it( "renders the component correctly for default props", () => {
		const { container } = render( <RouteLayout { ...props } /> );
		expect( container ).toMatchSnapshot();
	} );

	it( "renders the component correctly for custom props", () => {
		const { container } = render( <RouteLayout
			{ ...props }
			title="Custom title"
			description="custom description"
		/> );
		expect( container ).toMatchSnapshot();
	} );
} );
