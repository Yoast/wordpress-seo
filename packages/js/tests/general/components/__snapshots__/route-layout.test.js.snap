// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RouteLayout renders the component correctly for custom props 1`] = `
<div>
  <header
    class="yst-p-8 yst-border-b yst-border-slate-200"
  >
    <div
      class="yst-max-w-screen-sm"
    >
      <h1
        class="yst-title yst-title--1"
      >
        Custom title
      </h1>
      <p
        class="yst-text-tiny yst-mt-3"
      >
        custom description
      </p>
    </div>
  </header>
  empty component 
  <div>
    <div
      aria-live="assertive"
      role="log"
      style="border: 0px; height: 1px; margin: -1px; overflow: hidden; white-space: nowrap; padding: 0px; width: 1px; position: absolute;"
    />
    <div
      aria-live="assertive"
      role="log"
      style="border: 0px; height: 1px; margin: -1px; overflow: hidden; white-space: nowrap; padding: 0px; width: 1px; position: absolute;"
    />
    <div
      aria-live="polite"
      role="log"
      style="border: 0px; height: 1px; margin: -1px; overflow: hidden; white-space: nowrap; padding: 0px; width: 1px; position: absolute;"
    >
      Custom title Dashboard - Yoast SEO
    </div>
    <div
      aria-live="polite"
      role="log"
      style="border: 0px; height: 1px; margin: -1px; overflow: hidden; white-space: nowrap; padding: 0px; width: 1px; position: absolute;"
    />
  </div>
</div>
`;

exports[`RouteLayout renders the component correctly for default props 1`] = `
<div>
  <header
    class="yst-p-8 yst-border-b yst-border-slate-200"
  >
    <div
      class="yst-max-w-screen-sm"
    >
      <h1
        class="yst-title yst-title--1"
      >
        Title
      </h1>
      <p
        class="yst-text-tiny yst-mt-3"
      >
        Description
      </p>
    </div>
  </header>
  empty component 
  <div>
    <div
      aria-live="assertive"
      role="log"
      style="border: 0px; height: 1px; margin: -1px; overflow: hidden; white-space: nowrap; padding: 0px; width: 1px; position: absolute;"
    />
    <div
      aria-live="assertive"
      role="log"
      style="border: 0px; height: 1px; margin: -1px; overflow: hidden; white-space: nowrap; padding: 0px; width: 1px; position: absolute;"
    />
    <div
      aria-live="polite"
      role="log"
      style="border: 0px; height: 1px; margin: -1px; overflow: hidden; white-space: nowrap; padding: 0px; width: 1px; position: absolute;"
    >
      Title Dashboard - Yoast SEO
    </div>
    <div
      aria-live="polite"
      role="log"
      style="border: 0px; height: 1px; margin: -1px; overflow: hidden; white-space: nowrap; padding: 0px; width: 1px; position: absolute;"
    />
  </div>
</div>
`;
